### PR TITLE
FIX-#4808: Set dtypes correctly after column rename.

### LINKED ISSUE #4808: BUG: `df.describe` fails for DataFrame created with `add_prefix`
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: macOS 12.2.1
- **Modin version** (`modin.__version__`): 0.15.0+75.g3f985ed6
- **Python version**: 3.9.12
- **Code we can use to reproduce**:

```python
import modin.pandas as pd
import numpy as np

df = pd.DataFrame(np.random.randn(20, 16)).add_prefix('col')

# This fails
df.describe()
```

### Describe the problem
`df.describe()` fails on a DataFrame that has its columns set with `add_prefix`. 

### Source code / logs

<details>
<summary>Stack trace</summary>

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Input In [19], in <cell line: 1>()
----> 1 df.describe()

File ~/Documents/modin/modin/logging/logger_decorator.py:128, in enable_logging.<locals>.decorator.<locals>.run_and_log(*args, **kwargs)
    113 """
    114 Compute function with logging if Modin logging is enabled.
    115 
   (...)
    125 Any
    126 """
    127 if LogMode.get() == "disable":
--> 128     return obj(*args, **kwargs)
    130 logger = get_logger()
    131 logger_level = getattr(logger, log_level)

File ~/Documents/modin/modin/pandas/base.py:1172, in BasePandasDataset.describe(self, percentiles, include, exclude, datetime_is_numeric)
   1169 else:
   1170     percentiles = np.array([0.25, 0.5, 0.75])
   1171 return self.__constructor__(
-> 1172     query_compiler=self._query_compiler.describe(
   1173         percentiles=percentiles,
   1174         include=include,
   1175         exclude=exclude,
   1176         datetime_is_numeric=datetime_is_numeric,
   1177     )
   1178 )

File ~/Documents/modin/modin/logging/logger_decorator.py:128, in enable_logging.<locals>.decorator.<locals>.run_and_log(*args, **kwargs)
    113 """
    114 Compute function with logging if Modin logging is enabled.
    115 
   (...)
    125 Any
    126 """
    127 if LogMode.get() == "disable":
--> 128     return obj(*args, **kwargs)
    130 logger = get_logger()
    131 logger_level = getattr(logger, log_level)

File ~/Documents/modin/modin/core/storage_formats/pandas/query_compiler.py:1563, in PandasQueryCompiler.describe(self, **kwargs)
   1560 def describe(self, **kwargs):
   1561     # Use pandas to calculate the correct columns
   1562     empty_df = (
-> 1563         pandas.DataFrame(columns=self.columns)
   1564         .astype(self.dtypes)
   1565         .describe(**kwargs)
   1566     )
   1567     new_index = empty_df.index
   1569     # Note: `describe` convert timestamp type to object type
   1570     # which results in the loss of two values in index: `first` and `last`
   1571     # for empty DataFrame.

File ~/opt/anaconda3/envs/modin/lib/python3.9/site-packages/pandas/core/generic.py:5879, in NDFrame.astype(self, dtype, copy, errors)
   5877 for col_name in dtype.keys():
   5878     if col_name not in self:
-> 5879         raise KeyError(
   5880             "Only a column name can be used for the "
   5881             "key in a dtype mappings argument. "
   5882             f"'{col_name}' not found in columns."
   5883         )
   5885 # GH#44417 cast to Series so we can use .iat below, which will be
   5886 #  robust in case we
   5887 from pandas import Series

```

</details>

### PR DESCRIPTION
**Pull request by mvashishtha**
Signed-off-by: mvashishtha <mahesh@ponder.io>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4808
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Review by RehanSD — APPROVED** [review]
LGTM!

File: modin/pandas/test/dataframe/test_map_metadata.py [hunk 1]
```diff
@@ -205,6 +205,9 @@ def test_add_prefix(data):
     new_modin_df = modin_df.add_prefix(test_prefix)
     new_pandas_df = pandas_df.add_prefix(test_prefix)
     df_equals(new_modin_df.columns, new_pandas_df.columns)
+    # TODO(https://github.com/modin-project/modin/issues/3804):
+    # make df_equals always check dtypes.
+    df_equals(new_modin_df.dtypes, new_pandas_df.dtypes)
```
Comment (pyrito) [inline modin/pandas/test/dataframe/test_map_metadata.py:210]: Did you confirm that this test was failing before?

**Review by pyrito — APPROVED** [review]
Wooo! LGTM

File: modin/pandas/test/dataframe/test_map_metadata.py (hunk 1 above)
Comment (mvashishtha) [reply modin/pandas/test/dataframe/test_map_metadata.py:210]: Yep, it was.

File: modin/core/dataframe/pandas/dataframe/dataframe.py [hunk 2]
```diff
@@ -1754,13 +1754,19 @@ def map_fn(df):
             return df.rename(index=new_row_labels, columns=new_col_labels, level=level)
 
         new_parts = self._partition_mgr_cls.map_partitions(self._partitions, map_fn)
+
+        if self._dtypes is None:
+            new_dtypes = None
+        else:
+            new_dtypes = self._dtypes.set_axis(new_cols)
```
Comment (pyrito) [inline modin/core/dataframe/pandas/dataframe/dataframe.py:1761]: I guess one style nit (up to you if you wanna do this at this point):
```suggestion
        new_dtypes = None if self._dtypes is None else self._dtypes.set_axis(new_cols)
```

**Review by pyrito — APPROVED** [review]
LGTM (for real this time :D)

**Review by RehanSD — APPROVED** [review]
LGTM!

**Review by anmyachev — APPROVED** [review]

**Pull request merged**